src/chapters/spring/mvc.rst

```diff
@@ -0,0 +1,28 @@
+Design Patterns, MVC, and Spring, Oh My!
+========================================
+
+.. index:: ! design pattern, ! Model-View-Controller, ! MVC, ! model, ! view, ! controller
+
+So far, we have been designing our applications by diagrammming classes, drawing connections, and adding in interfaces.
+This practice benefits us because we can start seeing issues *before* we start coding.
+Many software developers start their applications with this process.
+But before we start diagramming our ``Cat`` class and our ``HouseCat`` class, we decide on the template for our design that we want to use.
+These design templates that are abstract solutions to common software architecture problems are called **design patterns**.
+Design patterns provide a set of conventions that we follow to build an application.
+
+**Model-View-Controller** (MVC) is a design pattern where the programming logic behind the application is broken down into 3 components: models, views, and controllers.
+A **model** handles the data and business logic of the application. A **view** handles the user interface elements.
+A **controller** passes information from the models to the views. Controllers are the "traffic cops" of the application, capable of passing data back and forth to the browser in MVC web applications. This process will be covered in depth later on in this chapter.
+
+.. TODO: Add figure showing the relationship between models, views, and controllers.
```

Comment:
did you see this diagram?/ do you have thoughts on it?

https://docs.google.com/presentation/d/1hcXmcVMUpjR3sLILtIHmY8XhtXddV_UgbFoOff-PpQ8/edit#slide=id.g1cc62e17bd_0_55